client/hostvolumemanager/host_volumes.go

```diff
@@ -84,11 +84,17 @@ func NewHostVolumeManager(logger hclog.Logger, config Config) *HostVolumeManager
 func (hvm *HostVolumeManager) Create(ctx context.Context,
 	req *cstructs.ClientHostVolumeCreateRequest) (*cstructs.ClientHostVolumeCreateResponse, error) {
 
+	log := hvm.log.With("volume_name", req.Name, "volume_id", req.ID)
+
 	plug, err := hvm.getPlugin(req.PluginID)
 	if err != nil {
 		return nil, err
 	}
 
+	// check if the volume already exists, so we can auto-delete on initial create
+	// (not update or restore) if we fail to save client state.
+	isNewVolume := !hvm.locker.isLocked(req.Name)
+
 	// can't have two of the same volume name w/ different IDs per client node
 	if err := hvm.locker.lock(req.Name, req.ID); err != nil {
 		return nil, err
```

Comment:
We might want to fold this check into `.lock` so that it returns `(bool, error)`. That way there's no risk of time-of-check-time-of-use issues with concurrent requests. And then you can call something like:

```go
isNewVolume, err := hvm.locker.lock(req.Name, req.ID)
if err != nil {
	return nil, err
}
```

Reply:
and done f031738c502157cd2d867e2f0366ced558d1be32